gabxgonz/cat-game-collab
# Falls and Breaks Script

![break-vase2](https://user-images.githubusercontent.com/21282705/134047458-8e41048e-4301-4686-8aa3-3bf0b1cbd714.gif)
# Testing

1. Add a new interactable object to the scene, preferably on top of some furniture.
2. Use Interactable tag on this object
3. Set layer as WhatIsInteractable
4. Make sure the floor stuff below (including rugs) is tagged as "Floor". This is the collision that will cause the item to break.
5. Add Falls and Breaks Script; it should add other required scripts
5. Rigidbody component: uncheck `Use Gravity` and check `Is Kinematic`
6. Box Collider component: check `Is Trigger`
7. Add Interaction Option to Interactable UI Script
9. Particle System component: Use the particle system preset & update render settings to use mesh(see screenshots)
10. Break it!

![Screen Shot 2021-09-20 at 9 34 37 AM](https://user-images.githubusercontent.com/21282705/134046154-d23b3a3f-5f24-45df-b041-bfa197d23f0e.png)

![image](https://user-images.githubusercontent.com/21282705/134047213-a8873084-2476-4861-947a-1d63c45ebc24.png)

![Screen Shot 2021-09-20 at 9 35 05 AM](https://user-images.githubusercontent.com/21282705/134046147-63f3889d-1323-4a03-a721-cb8ebf746d1a.png)

![Screen Shot 2021-09-20 at 9 35 26 AM](https://user-images.githubusercontent.com/21282705/134046134-286e5f2c-5a72-4c20-97af-79ba8428e923.png)

# Notes
- I feel like `handleInteractablity()` is probably a nightmare.
